modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/h2/twostep/GridReduceQueryExecutor.java

```diff
@@ -634,8 +640,9 @@ else if (msg.page() == 0) {
 
                 boolean retry = false;
 
-                // Always enforce join order on map side to have consistent behavior.
-                int flags = GridH2QueryRequest.FLAG_ENFORCE_JOIN_ORDER;
+                int flags = singlePartMode ?
```

Comment:
Looks like an optimization for single certain use case. Would next be more readable?
singlePartMode && !enforceJoinOrder ? 0 : FLAG_....

Reply:
fixed.